themes/purity/purity.theme.bash

```diff
@@ -14,9 +14,20 @@ STATUS_THEME_PROMPT_BAD="${bold_red}❯${reset_color}${normal} "
 STATUS_THEME_PROMPT_OK="${bold_green}❯${reset_color}${normal} "
 PURITY_THEME_PROMPT_COLOR="${PURITY_THEME_PROMPT_COLOR:=$blue}"
 
+venv_prompt() {
+	python_venv=""
+	# Detect python venv
+	if [[ -n "${CONDA_DEFAULT_ENV}" ]]; then
+		python_venv="($PYTHON_VENV_CHAR${CONDA_DEFAULT_ENV}) "
+	elif [[ -n "${VIRTUAL_ENV}" ]]; then
+		python_venv="($PYTHON_VENV_CHAR$(basename "${VIRTUAL_ENV}")) "
+	fi
+	[[ -n "${python_venv}" ]] && echo "${python_venv}"
+}
+
 function prompt_command() {
     local ret_status="$( [ $? -eq 0 ] && echo -e "$STATUS_THEME_PROMPT_OK" || echo -e "$STATUS_THEME_PROMPT_BAD")"
```

Comment:
Shellcheck gives [this](https://github.com/koalaman/shellcheck/wiki/SC2181) warning on this line.
I can workaround it by:
`retval=$?`
`local ret_status="$([ $retval -eq 0 ] && echo -e "$STATUS_THEME_PROMPT_OK" || echo -e "$STATUS_THEME_PROMPT_BAD")"`

Since this is supposed to check the return value of any command run previously, is there a better way to do it without raising the shellchek warning?

Reply:
Hey @tiagobarros99,

Your idea is the right direction.  Here's how the [powerline theme](https://github.com/Bash-it/bash-it/blob/master/themes/powerline/powerline.base.bash#L251) does it:
```
function __powerline_prompt_command() {
	local last_status="$?" ## always the first [action]
        ...
	[[ "${last_status}" -ne 0 ]] && __powerline_left_segment "$(__powerline_last_status_prompt ${last_status})"
        ...
}
```

So I think its considered a good pattern to capture `$?` immediately in a local variable and then use the variable to drive logic.